### PR TITLE
Add unit tests for new matchers 'shouldNotBePositive', 'shouldNotBeNe…

### PR DESCRIPTION
**Pull request by raverona**
…gative' and unit tests for both

Fixes: #435

### CODE REVIEW

File: kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    row(Double.NEGATIVE_INFINITY)
+            )
+
+            forAll(table) {
+                it.shouldNotBePositive()
+                it shouldNotBe positive()
+            }
+
+            shouldThrow<AssertionError> {
+                0.1.shouldNotBePositive()
```
Comment (Kerooker) [inline kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt]: This will stop tests

Comment (raverona) [reply kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt]: ok, will change it

**Comment by raverona** [comment]
Actually I discovered a bigger issue about this PR then the one with the tests. The thing is: should `Double.NaN` be considered not positive? (and not negative?). There are two possible interpretations: 

1. no, because NaN is not a number, so it does not make sense to say it's positive or negative or not positive or not negative.
2. yes, because NaN > 0 is false and not positive is the negation of that

if we follow the current implementation the interpretation in this case would be the second one, but which one is the correct or most accurate?

**Comment by Kerooker** [comment]
A 3rd option to this would be "Don't consider anything, throw an exception."

If an user tries to assert that NaN is either positive or negative, don't let them do this. It shouldn't be an assertion at all.

**Comment by raverona** [comment]
I pushed a commit that correct the tests according to the current implementation of the `positive` and `negative` functions, so this PR can be merged if its ok to expect this behavior (answering yes for `Double.NaN.shouldNotBePositive()`) when asking if `Double.NaN` is not positive and not negative

File: doc/matchers.md [hunk 2]
```diff
@@ -90,6 +90,10 @@ For the extension function style, each function has an equivalent negated versio
 | `double.shouldBeLessThanOrEqual(n)` | Asserts that the double is less or equal to than the given value n |
 | `double.shouldBeGreaterThan(n)` | Asserts that the double is greater than the given value n |
 | `double.shouldBeGreaterThanOrEqual(n)` | Asserts that the double is greater than or equal to the given value n |
+| `double.shouldBePositive()` | Asserts that the double is positive |
+| `double.shouldBeNotPositive()` | Asserts that the double is not positive|
```
Comment (Kerooker) [inline doc/matchers.md]: The documentation here is wrong. I believe you're aiming for `double.shouldNotBePositive`

File: doc/matchers.md [hunk 3]
```diff
@@ -90,6 +90,10 @@ For the extension function style, each function has an equivalent negated versio
...
+| `double.shouldBeNegative()` | Asserts that the double is negative |
+| `double.shouldBeNotNegative()` | Asserts that the double is not negative |
```
Comment (Kerooker) [inline doc/matchers.md]: Same as the positive case, I believe you're going for `double.shouldNotBeNegative`

Comment (raverona) [reply doc/matchers.md]: oh that's correct, my bad

**Comment by sksamuel** [comment]
:+1:

**Pull request merged**